# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ShareableCanvasRenderer.cpp
# Commit: 0b191b707750
# Full Hash: 0b191b707750e040289033359306e4966c1bf062
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r 16c5dd2cfd60 -r 0b191b707750 gfx/layers/ShareableCanvasRenderer.cpp
--- a/gfx/layers/ShareableCanvasRenderer.cpp	Mon Jun 15 18:25:55 2020 +0000
+++ b/gfx/layers/ShareableCanvasRenderer.cpp	Mon Jun 15 18:25:58 2020 +0000
@@ -11,6 +11,7 @@
 #include "mozilla/layers/TextureClientSharedSurface.h"
 #include "mozilla/layers/CompositableForwarder.h"
 
+#include "ClientWebGLContext.h"
 #include "gfxUtils.h"
 #include "GLScreenBuffer.h"
 #include "nsICanvasRenderingContextInternal.h"
@@ -122,7 +123,12 @@
 
     if (!webgl) return nullptr;
     if (!forwarder) return nullptr;
-    const auto desc = webgl::GetFrontBuffer(*webgl, forwarder);
+
+    auto texType = layers::TextureType::Unknown;
+    if (forwarder) {
+      texType = layers::PreferredCanvasTextureType(*forwarder);
+    }
+    const auto desc = webgl->GetFrontBuffer(nullptr, texType);
     if (!desc) return nullptr;
     return GetFrontBufferFromDesc(*desc, flags);
   };